router/router.go

```diff
@@ -161,155 +159,108 @@ func (rt *HandleT) workerProcess(worker *workerT) {
+		if toBackoff {
+			continue
 		}
 
-		var reqMetric requestMetric
-		diagnosisStartTime := time.Now()
-		//We can execute thoe job
-		for attempts = 0; attempts < ser; attempts++ {
-			logger.Debugf("[%v Router] :: trying to send payload. Attempt no. %v of max attempts %v", rt.destID, attempts, ser)
+		// mark job as throttled if either dest level event limit reached or dest user level limit reached
+		hasBeenThrottled := rt.throttle(job, paramaters, userID, worker, isPrevFailedUser)
+		if hasBeenThrottled {
+			continue
+		}
 
-			deliveryTimeStat.Start()
+		// START: request to destiantion endpoint
 
-			if rt.customDestinationManager != nil {
-				respStatusCode, respStatus, respBody = rt.customDestinationManager.SendData(job.EventPayload, paramaters.SourceID, paramaters.DestinationID)
-			} else {
-				respStatusCode, respStatus, respBody = rt.netHandle.sendPost(job.EventPayload)
-			}
+		logger.Debugf("[%v Router] :: trying to send payload. Attempt no. %v of max attempts %v", rt.destName, attempts, ser)
 
-			deliveryTimeStat.End()
+		deliveryTimeStat.Start()
 
-			if useTestSink {
-				//Internal test. No reason to sleep
-				break
-			}
-			if !isSuccessStatus(respStatusCode) {
-				reqMetric.RequestRetries = reqMetric.RequestRetries + 1
-				if attempts == ser-1 {
-					reqMetric.RequestAborted = reqMetric.RequestAborted + 1
-					reqMetric.RequestCompletedTime = time.Now().Sub(diagnosisStartTime)
-				}
-				//400 series error are client errors. Can't continue
-				if respStatusCode >= http.StatusBadRequest && respStatusCode <= http.StatusUnavailableForLegalReasons {
-					break
-				}
-				//Wait before the next retry
-				worker.sleepTime = 2*worker.sleepTime + 1*time.Second //+1 handles 0 sleepTime
-				if worker.sleepTime > maxSleep {
-					worker.sleepTime = maxSleep
-				}
-				logger.Debugf("[%v Router] :: worker %v sleeping for  %v ",
-					rt.destID, worker.workerID, worker.sleepTime)
-				time.Sleep(worker.sleepTime)
+		if job.LastJobStatus.AttemptNum > 0 {
+			retryAttemptsStat.Increment()
+		}
+
+		if rt.customDestinationManager != nil {
+			respStatusCode, _, respBody = rt.customDestinationManager.SendData(job.EventPayload, paramaters.SourceID, paramaters.DestinationID)
+		} else {
+			respStatusCode, _, respBody = rt.netHandle.sendPost(job.EventPayload)
+		}
 
-				retryAttemptsStat.Increment()
+		deliveryTimeStat.End()
 
-				continue
-			} else {
-				atomic.AddUint64(&rt.successCount, 1)
-				//Divide the sleep
-				worker.sleepTime = worker.sleepTime / 2
-				if worker.sleepTime < minSleep {
-					worker.sleepTime = minSleep
-				}
-				logger.Debugf("[%v Router] :: sleep for worker %v decreased to %v",
-					rt.destID, worker.workerID, worker.sleepTime)
-				// stop time - success
-				reqMetric.RequestSuccess = reqMetric.RequestSuccess + 1
-				reqMetric.RequestCompletedTime = time.Now().Sub(diagnosisStartTime)
-				break
-			}
+		// END: request to destiantion endpoint
+
+		if useTestSink {
```

Comment:
Let's remove useTestSink too from router. Our tests changed recently and we don't use this anymore.